zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -389,6 +389,20 @@ impl<'a, F: FieldExt> ConstraintBuilder<'a, F> {
         );
     }
 
+    // block
+    pub(crate) fn block_lookup(
+        &mut self,
+        tag: Expression<F>,
+        number: Option<Expression<F>>,
+        val: Expression<F>,
+    ) {
+        self.add_lookup(Lookup::Block {
+            field_tag: tag,
+            number: number.unwrap_or_else(|| 0.expr()),
```

Comment:
If the block comes without a number provided shouldn't we panic directly? As it means that some deep internal logic must be wrong.

Reply:
if the `block` construct not initialize explicitly with a number, the default value will set to zero,  so will not panic directly, does it make sense?

Reply:
But 0 would probbaly not be the correct value no? So we will be passing a wrong block to the circuit condtructor IIUC.

Reply:
the block constants include block number will be pub input of circuit eventually as planned, so it will be easy to figure out if it is incorrect .

Reply:
Yes, what I meant is that if you check the code, if there's any unexpected error/behavior and we receive a `None/Err`, putting a 0 will not be correct. 

Therefore the question of paniking.

Reply:
Ah, that is it , at first I used `number` parameter type `Expression` directly without `Option` wrapper, but considering maybe zero in cases  , change to `Option` as now.  what is your ideas ?

Reply:
To me it looks like `number` should not be an `Option` or `Result` and instead a pure value directly. 
In that case, it's impossible to construct a non-valid `number`.

Maybe @han0110 has some insight on that. I'm just wondering that this cannot be the source of errors in the future. If it cannot, we can simply resolve the conversation.

Reply:
Then you can proceed as you wish!!!